packages/core/src/plugins/use-masked-errors.ts

```diff
@@ -0,0 +1,65 @@
+import { Plugin, ExecutionResult } from '@envelop/types';
+import { handleStreamOrSingleExecutionResult } from '../utils.js';
+
+export const DEFAULT_ERROR_MESSAGE = 'Unexpected error.';
+
+export type FormatErrorHandler = (error: unknown, message: string) => Error;
+
+export const formatError: FormatErrorHandler = (err, message) => {
+  if (err?.name === 'GraphQLError') {
+    if (err?.originalError) {
+      if (err.originalError.name === 'GraphQLError') {
+        return err as Error;
+      }
+      return new Error(message);
+    }
+    return err as Error;
+  }
+  return new Error(message);
+};
```

Comment:
This misses the `originalError` extensions for dev: - https://github.com/n1ru4l/envelop/blob/cea2df214e4d598f1206b1e066bf285960011378/packages/core/src/plugins/use-masked-errors.ts#L30-L37
- https://www.the-guild.dev/graphql/yoga-server/docs/features/error-masking#receive-original-error-in-development

I would prefer to keep this behavior as it can be quite handy.

Reply:
Regular error object doesn’t really have any format for additional metadata

Reply:
I think this is highly confusing if the error is suddenly an Error instance instead of an "GraphQLError", that means it is also impossible to add extensions to an masked error? That is a weird limitation as the `formatError` function should imho always return a GraphQL conform JSON serializable error.

Reply:
```bash
node
> JSON.stringify(new Error("lol"))
'{}'
```

this would break `JSON.stringify()` of an execution result.

Reply:
Maybe we cannot get around a custom class...

Reply:
Yeah the other strategy is we pull this out from core and provide a plugin for implementation users are using